pkg/controllers/monitoring/monitoring-stack/controller.go

```diff
@@ -98,14 +116,18 @@ func RegisterWithManager(mgr ctrl.Manager, opts Options) error {
 	ctrl, err := ctrl.NewControllerManagedBy(mgr).
 		WithLogger(ctrl.Log).
 		For(&stack.MonitoringStack{}).
-		Owns(&monv1.Prometheus{}).WithEventFilter(p).
+		Owns(&monv1.Prometheus{}).
 		Owns(&monv1.Alertmanager{}).WithEventFilter(p).
 		Owns(&v1.Service{}).WithEventFilter(p).
 		Owns(&v1.ServiceAccount{}).WithEventFilter(p).
 		Owns(&rbacv1.Role{}).WithEventFilter(p).
 		Owns(&rbacv1.RoleBinding{}).WithEventFilter(p).
 		Owns(&monv1.ServiceMonitor{}).WithEventFilter(p).
 		Owns(&policyv1.PodDisruptionBudget{}).WithEventFilter(p).
+		Watches(
+			&source.Kind{Type: &monv1.Prometheus{}},
```

Comment:
Not quite sure I understand this part. I might be missing something, but isn't  `Owns(&monv1.Prometheus{}).` already responsible for triggering reconciliation when Prometheus CRs are updated?

Reply:
Yes. Correct. This was just updated. Sorry. However I will raise another question here.